cmd/api-linter/cli.go

```diff
@@ -105,14 +111,33 @@ func runCLI(rules lint.Rules, configs lint.Configs, args []string) error {
 				case "json":
 					marshal = json.Marshal
 				}
-				b, err := marshal(problems)
+				b, err := marshal(lintResponses)
 				if err != nil {
 					return err
 				}
-
 				if _, err = w.Write(b); err != nil {
 					return err
 				}
+
+				if c.BoolT("summary") {
+					summary, longestRuleLen := createSummary(lintResponses)
+					colOneFormat, colTwoFormat := "%-"+strconv.Itoa(max(longestRuleLen+5, 25))+"s", "%25s"
+					w.WriteString("\n----------SUMMARY TABLE---------\n")
+					w.WriteString(fmt.Sprintf(colOneFormat,
+						fmt.Sprintf("Linted %d proto files.", len(lintResponses))) + "\n")
+					w.WriteString(fmt.Sprintf(colOneFormat, "Rule") +
+						fmt.Sprintf(colTwoFormat, "Violations (Percent)") + "\n")
+
+					for rule_id, numViolations := range summary {
+						w.WriteString(fmt.Sprintf(colOneFormat, string(rule_id)) +
```

Comment:
I don't actually know off the top of my head if it there's any advantage/disadvantage to concatenating strings (+) vs multiple calls to w.WriteString.

Reply:
interesting. I couldn't find it online so I just did a test now with some big amount of strings. seems like writeString is  ~2x faster

Reply:
Nice find. Thx for looking into it.